go-controller/pkg/ovn/controller/apbroute/network_client.go

```diff
@@ -174,8 +182,10 @@ func (nb *northBoundClient) addGatewayIPs(pod *v1.Pod, egress *gateway_info.Gate
 		podIPs = append(podIPs, ipNet)
 	}
 	if len(podIPs) == 0 {
+		// At this stage the pod is either in Pending or Running phase, but Pending should not have an IP, therefore it should not
+		// be processed yet. Return an error to prevent the pod from being perceived as correctly configured with the gateway IP.
```

Comment:
i think the comment needs to be updated here because you are no longer returning an error and now returning applied or not as bool

Reply:
Done.